ui/desktop/electron-app/src/ipc/handlers.js

```diff
@@ -127,19 +127,36 @@ ipcMain.on('createTerminal', (event, payload) => {
     cwd: process.env.HOME,
     env: process.env,
   });
+  const incomingDataChannel = `terminalIncomingData-${id}`;
+  const keystrokeChannel = `terminalKeystroke-${id}`;
+  const resizeChannel = `resize-${id}`;
+  const removeChannel = `removeTerminal-${id}`;
 
   // This sends to the renderer and xterm whatever the ptyProcess (host terminal) outputs.
   ptyProcess.on('data', function (data) {
-    sender.send(`terminalIncomingData-${id}`, data);
+    sender.send(incomingDataChannel, data);
   });
 
   // This writes into ptyProcess (host terminal) whatever we write through xterm.
-  ipcMain.on(`terminalKeystroke-${id}`, (event, value) => {
+  ipcMain.on(keystrokeChannel, (event, value) => {
     ptyProcess.write(value);
   });
 
-  // Resize the number of columns and rows received from xterm
-  ipcMain.on(`resize-${id}`, (event, { cols, rows }) => {
+  // Resize the number of columns and rows received from xterm.
+  ipcMain.on(resizeChannel, (event, { cols, rows }) => {
     ptyProcess.resize(cols, rows);
   });
+
+  // We use `ipcMain.once` as we want this listener cleaned up
+  // after killing the ptyProcess as well.
+  ipcMain.once(removeChannel, () => {
+    // Just let the error bubble up since we can't do anything
+    try {
+      ptyProcess.kill();
```

Comment:
Should we do anything if there's an error killing the process?

Reply:
@ZedLi 
That's actually a pretty good question. What we can do? I don't think we need to report this to the user. But if this can cause problems with other terminals, maybe we can force a refresh of the app?

Reply:
I like @calcaide's idea of force refreshing the app in case of multiple terminals.

Reply:
Wouldn't that be too disruptive for the user? Does restarting the app guarantee the killing of the spawned processes?